common-tasks/usb.md

```diff
@@ -17,6 +17,160 @@ redirect_from:
+approaches to this problem:
+
+1. Physically disconnect all USB devices whenever you reboot the host.
+2. Hide (i.e., blacklist) all USB controllers from dom0.
+
+**Warning:** If you use a USB [AEM] device, do not use the second option. Using
+a USB AEM device requires dom0 to have access to the USB controller to which
+your USB AEM device is attached. If dom0 cannot read your USB AEM device, AEM
+will hang.
+
+The procedure to hide all USB controllers from dom0 is as follows:
+
+1. Open the file `/etc/default/grub` in dom0.
+2. Find the line that begins with `GRUB_CMDLINE_LINUX`.
+3. Add `rd.qubes.hide_all_usb` to that line.
+4. Save and close the file.
+5. Run the command `grub2-mkconfig -o /boot/grub2/grub.cfg` in dom0.
+6. Reboot.
+
+(Note: Beginning with R3.2, `rd.qubes.hide_all_usb` is set automatically if you
+opt to create a USB qube during installation. This also occurs automatically if
+you choose to [create a USB qube] using the `qubesctl` method, which is the
+first pair of steps in the linked section.)
+
+**Warning** USB keyboard cannot be used to type the disk passphrase
+if USB controllers were hidden from dom0. Before hiding USB controllers
+make sure your laptop keyboard is not internally connected via USB
+(by checking output of `lsusb` command) or that you have a PS/2 keyboard at hand
+(if using a desktop PC). Failure to do so will render your system unusable.
+
+
+Removing a USB qube
+-------------------
+
+**Warning:** This procedure will result in your USB controller(s) being attached
+directly to dom0.
+
+1. Shut down the USB qube.
+2. In Qubes Manager, right-click on the USB qube and select "Remove VM."
+3. Open the file `/etc/default/grub` in dom0.
+4. Find the line(s) that begins with `GRUB_CMDLINE_LINUX`.
+5. If `rd.qubes.hide_all_usb` appears anywhere in those lines, remove it.
+6. Save and close the file.
+7. Run the command `grub2-mkconfig -o /boot/grub2/grub.cfg` in dom0.
+8. Reboot.
+
+How to use a USB keyboard
+-------------------------
+
+In order to use a USB keyboard, you must first attach it to a USB qube, then
+give that qube permission to pass keyboard input to dom0. Note that allowing
+keyboard access from a USB qube gives it great power. In short:
+
+ * It will see whatever you type on that keyboard (including passwords).
+ * It will be able to inject keystrokes, which basically means that it will be
+   able to enter any command. For example, if some malware catches your
+   screenlocker password, it will be able to unlock the screen when you are not
+   present. (For more details, see [here][input-proxy].)
+
+If you are sure you wish to proceed, then you must edit the
+`qubes.InputKeyboard` policy file in dom0, which is located here:
+
+    /etc/qubes-rpc/policy/qubes.InputKeyboard
+
+Add a line like this one to the top of the file:
+
+    sys-usb dom0 ask,user=root
+
+(Change `sys-usb` to your desired USB qube.)
+
+You can now use your USB keyboard.
+
+How to use a USB mouse
```

Comment:
> So, in this case, I'm in favor of just linking to one place with such warning, instead of replicating it everywhere. So, the question is where should be this place - here, or in https://github.com/QubesOS/qubes-app-linux-input-proxy/blob/master/README.md?

I think it would be better to have it in the docs (so that it will be included if we ever have offline docs, or if someone clones qubes-doc for their own use).

In the interest of time, I'll merge this PR and update the warnings in follow-up commits.